packages/server/src/tests/header/headerSelection.ts

```diff
@@ -107,91 +91,96 @@ const supportersTest: HeaderTest = {
-    return isAfter(date, usEoyPeriodStart) && isBefore(date, usEoyPeriodEnd);
-};
+const selectBestTest = (
+    targeting: HeaderTargeting,
+    allTests: HeaderTest[],
+): HeaderTestSelection => {
+    // If there is a need to return other specific hardcoded tests/variants, include required logic to identify, and required return, here
 
-const hasNotContributedDuringUsEoyPeriod = (lastOneOffContributionDate?: string): boolean => {
-    if (!lastOneOffContributionDate) {
-        return true;
+    // In case the allTests array is empty, return a hardcoded test/variant
+    if (!allTests.length) {
+        return selectHardcodedTest(targeting);
     }
 
-    return isBefore(new Date(Date.parse(lastOneOffContributionDate)), usEoyPeriodStart);
-};
+    // Cascade through allTests to match supplied targeting data to the first test that meets requirements - if no test meets requirements, return an appropriate hardcoded test/variant
+    let selectedTest;
+
+    const { showSupportMessaging, countryCode } = targeting;
+
+    const countryGroupId = countryCodeToCountryGroupId(countryCode.toUpperCase());
+
+    for (let i = 0, iz = allTests.length; i < iz; i++) {
+        const test = allTests[i];
+
+        const { isOn, userCohort, locations } = test;
 
-const getNonSupportersTest = (edition: Edition): HeaderTest => {
-    if (edition === 'UK') {
-        return nonSupportersTestUK;
+        if (isOn) {
+            if (showSupportMessaging && existingSupportersAndEveryone.indexOf(userCohort) >= 0) {
+                if (locations.indexOf(countryGroupId) >= 0) {
+                    selectedTest = test;
+                    break;
+                }
+            } else {
+                if (locations.indexOf(countryGroupId) >= 0) {
+                    selectedTest = test;
+                    break;
+                }
+            }
+        }
     }
-    if (edition === 'US' && isInUsEoyPeriod(new Date())) {
-        return nonSupportersTestUS();
+    // Failed to find a matching test, or the matching test has an empty variants Array
+    if (!selectedTest || !selectedTest.variants.length) {
+        return selectHardcodedTest(targeting);
     }
-    return nonSupportersTestNonUK;
+
+    // Identify the required variant
+    const selectedVariant: HeaderVariant = selectVariant(selectedTest, targeting.mvtId);
+
+    // Return the selected test and variant
+    selectedVariant.modulePathBuilder = modulePathBuilder;
+
+    return {
+        test: selectedTest,
+        variant: selectedVariant,
+        moduleName: header.name,
+        modulePathBuilder,
+    };
 };
 
-const getSupportersTest = (edition: Edition, lastOneOffContributionDate?: string): HeaderTest => {
-    if (
-        edition === 'US' &&
-        isInUsEoyPeriod(new Date()) &&
-        hasNotContributedDuringUsEoyPeriod(lastOneOffContributionDate)
-    ) {
-        return supportersTestUS;
+export const selectHardcodedTest = (targeting: HeaderTargeting): HeaderTestSelection => {
```

Comment:
instead of having a separate function for hardcoded tests we can construct an array of all tests, like we do for epics.
E.g.
```
const hardcodedTests = [nonSupportersTestNonUK, nonSupportersTestUK, supportersTest];

export const selectHeaderTest = async (
    targeting: HeaderTargeting,
): Promise<HeaderTestSelection | null> => {
    const configuredTests = await fetchConfiguredHeaderTestsCached().catch(() => []);
    return selectBestTest(targeting, [...configuredTests, ...hardcodedTests]);
};
```

Reply:
Fixed